oracle.py

```diff
@@ -73,20 +75,22 @@ def insert_card(self, name, card):
     sql += ', '.join('?' for prop in Oracle.properties())
     sql += ')'
     values = [card.get(self.underscore2camel(property)) for property in Oracle.properties()]
-    self.database.execute(sql, values)
+    # self.database.execute commits after each statement, which we want to 
+    # avoid while inserting cards
+    self.database.database.execute(sql, values)
     id = self.database.value('SELECT last_insert_rowid()')
     for name in card.get('names', []):
-      self.database.execute('INSERT INTO card_name (card_id, name) VALUES (?, ?)', [id, name])
+      self.database.database.execute('INSERT INTO card_name (card_id, name) VALUES (?, ?)', [id, name])
```

Comment:
Good God do I hate self.database.database :)

I feel like we're reaching into the internals of Database here. It might be nicer to build up an array of arrays of args and then process them all at the end? sqlite3 exposes executeMany which may or may not be useful.

If that's too much surgery then perhaps we can just expose a version of execute that doesn't commit or have a boolean param on execute for commit-or-not?